editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/create/ext/CreateFigureFeature.java

```diff
@@ -206,7 +261,20 @@ protected String getIconName() {
 			selectedContainer = (AbstractCamelModelElement)getBusinessObjectForPictogramElement(container);
 		}
 
-		AbstractCamelModelElement node = createNode(selectedContainer, selectedContainer != null);
+		CreateContext ctxNew = null;
```

Comment:
I have no exact solution here but this method is starting to be quite complicated to understand. Maybe we can try to extract different method?